source/Octopus.Tentacle.Tests.Integration/Util/RandomTemporaryFileBuilder.cs

```diff
@@ -38,7 +39,14 @@ public void Dispose()
         {
             if (File.Exists)
             {
-                File.Delete();
+                try
+                {
+                    _ = Task.Run(() => File.Delete());
+                }
+                catch
+                {
+                    // We really don't care
+                }
```

Comment:
We had some tests regularly failing, as the temp file was still in use.

After a code review of the test clean up logic in these tests, everything seemed to be getting cleaned up as expected, but the file is still _sometimes_ in use.

As this is class purely for testing, and we regularly recycle our build agents, we are happy with a 'best effort' approach here.

Reply:
Nit: This is a great explanation. It would be awesome to have an abbreviated version as the comment in code, as I think it would be a better comment.